File: frontend/src/api/recommendations.ts
/**
 * recommendations.ts — API client for Phase 5 ML Recommendations
 */
import { api } from "./client";

export interface ItemRecommendation {
  item: string;
  priority_score: number;
  reason: string;
  priority: "high" | "medium" | "low";
  confidence: number;
  source: string;
}

export interface SessionPlan {
  items: string[];
  session_length_s: number;
  focus_mode: "standard" | "deep_focus" | "confidence_boost" | "quick_win";
  reasoning: string;
  difficulty: string;
  confidence: number;
  source: string;
  learning_style: string;
  session_config: string;
}

export interface FullRecommendationResponse {
  learning_style: string;
  style_label: string;
  style_recommendation: string;
  style_confidence: number;
  style_source: string;
  recommended_difficulty: string;
  difficulty_confidence: number;
  difficulty_source: string;
  recommended_items: ItemRecommendation[];
  session_plan: SessionPlan;
  data_completeness: number;
  overall_confidence: number;
  source: string;
}

export interface RecommendationFeedbackRequest {
  recommendation_id?: string;
  item: string;
  action: "accepted" | "skipped" | "completed";
  outcome_accuracy?: number;
  module_type?: string;
}

export interface RecommendationFeedbackResponse {
  stored: boolean;
  message: string;
}

export const recommendationsApi = {
  full: (moduleType: string = "alphabet_practice") =>
    api.get<FullRecommendationResponse>(`/recommendations/full?module_type=${moduleType}`),
  
  session: (moduleType: string = "alphabet_practice") =>
    api.get<SessionPlan>(`/recommendations/session?module_type=${moduleType}`),

  feedback: (data: RecommendationFeedbackRequest) =>
    api.post<RecommendationFeedbackResponse>("/recommendations/feedback", data),
};

File: app/schemas/recommendations.py
"""app/schemas/recommendations.py — Recommendation DTOs"""

from pydantic import BaseModel
from typing import Literal


class ItemRecommendation(BaseModel):
    item:           str
    priority_score: float
    reason:         str
    priority:       Literal["high", "medium", "low"]
    confidence:     float
    source:         str


class SessionPlan(BaseModel):
    items:            list[str]
    session_length_s: int
    focus_mode:       Literal["standard", "deep_focus", "confidence_boost", "quick_win"]
    reasoning:        str
    difficulty:       str
    confidence:       float
    source:           str
    learning_style:   str
    session_config:   str
    audio_metadata:   dict | None = None


class FullRecommendationResponse(BaseModel):
    learning_style:          str
    style_label:             str
    style_recommendation:    str
    style_confidence:        float
    style_source:            str
    recommended_difficulty:  str
    difficulty_confidence:   float
    difficulty_source:       str
    recommended_items:       list[ItemRecommendation]
    session_plan:            SessionPlan
    data_completeness:       float
    overall_confidence:      float
    source:                  str


class RecommendationFeedbackRequest(BaseModel):
    recommendation_id: str | None = None
    item:              str
    action:            Literal["accepted", "skipped", "completed"]
    outcome_accuracy:  float | None = None   # accuracy achieved if completed
    module_type:       str = "alphabet_practice"


class RecommendationFeedbackResponse(BaseModel):
    stored:  bool
    message: str
